sharball-release.sh

```diff
@@ -0,0 +1,139 @@
+#!/bin/bash
+
+# We don't need return codes for "$(command)", only stdout is needed.
+# Allow `[[ -n "$(command)" ]]`, `func "$(command)"`, pipes, etc.
+# shellcheck disable=SC2312
+
+set -u
+
+abort() {
+  printf "%s\n" "$@"
+  exit 1
+}
+
+# Fail fast with a concise message when not using bash
+# Single brackets are needed here for POSIX compatibility
+# shellcheck disable=SC2292
+if [ -z "${BASH_VERSION:-}" ]
+then
+  abort "Bash is required to interpret this script."
+fi
+
+# string formatters
+if [[ -t 1 ]]
+then
+  tty_escape() { printf "\033[%sm" "$1"; }
+else
+  tty_escape() { :; }
+fi
+tty_mkbold() { tty_escape "1;$1"; }
+tty_blue="$(tty_mkbold 34)"
+tty_bold="$(tty_mkbold 39)"
+tty_reset="$(tty_escape 0)"
+
+shell_join() {
+  local arg
+  printf "%s" "$1"
+  shift
+  for arg in "$@"
+  do
+    printf " "
+    printf "%s" "${arg// /\ }"
+  done
+}
+
+ohai() {
+  printf "${tty_blue}==>${tty_bold} %s${tty_reset}\n" "$(shell_join "$@")"
+}
+
+line_range_of() {
+  head -n +"$3" "$1" | tail -n +"$2"
+}
+
+TMP_DIR="$(mktemp -d -t brew-install.XXXXXX)"
+BREW_DIR="${TMP_DIR}/brew"
+trap 'rm -rf "${TMP_DIR}"' EXIT
+
+HOMEBREW_BREW_GIT_REMOTE="https://github.com/Homebrew/brew.git"
+ohai "Creating fresh clone of Homebrew/brew from ${HOMEBREW_BREW_GIT_REMOTE}..."
+git clone https://github.com/Homebrew/brew.git "${BREW_DIR}"
```

Comment:
What about downloading the release tarball instead of cloning?

Reply:
I'm considering including the `.git` directory, which the release tarball does not have. Besides, it's easier to get the latest tag with a complete clone.

Reply:
I don't think we should be including a Git repository here.

Reply:
In the current installer `install.sh`, it always requires a Internet connection for `git fetch`. I think the sharball would be a offine installer. If we don't include a Git repository, what's the difference to use the online installer `install.sh`.

Reply:
Homebrew is usable without a Git repository. It's pretty atypical for software to bundle Git repositories. This would still allow you to download and install offline without including a Git repository.

Reply:
Note the primary goal is the DMG and secondary is the sharball which is another reason for using something like CPack that can generate both.